staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/conversion.go

```diff
@@ -62,12 +62,11 @@ func ConvertJSONSchemaPropsToOpenAPIv2Schema(in *apiextensions.JSONSchemaProps)
 			}
 		}
 
-		if len(p.Type) > 1 {
-			// https://github.com/kubernetes/kube-openapi/pull/143/files#diff-62afddb578e9db18fb32ffb6b7802d92R272
-			// We also set Properties to null to enforce parseArbitrary at https://github.com/kubernetes/kube-openapi/blob/814a8073653e40e0e324205d093770d4e7bb811f/pkg/util/proto/document.go#L247
+		switch {
+		case len(p.Type) == 2 && (p.Type[0] == "null" || p.Type[1] == "null"):
```

Comment:
do we have a test that `type:["object","null"], properties: ...` behaves correctly? (the only difference between this and the previous behavior is that properties is preserved, right?)

Reply:
It lets us go here: https://github.com/kubernetes/kube-openapi/blob/master/pkg/util/proto/document.go#L238. So it is handled the same way as `
type:"object", properties: ..` for the non-null case. The null case is always accepted in kube-openapi validation as far as I see, so nullable=true is implied.

I have started a follow-up TODO list in the description. I would like to have e2e tests with CRDs for these cases.